scripts/test.sh

```diff
@@ -342,12 +342,10 @@ function shellcheck_pass {
 function shellws_pass {
   TAB=$'\t'
   log_callout "Ensuring no tab-based indention in shell scripts"
-  local files
-  files=$(find ./ -name '*.sh' -print0 | xargs -0 )
-  # shellcheck disable=SC2206
-  files=( ${files[@]} "./scripts/build-binary.sh" "./scripts/build-docker.sh" "./scripts/release.sh" )
```

Comment:
First thought: We could, but we would also need to change lines 349 and 351, as they assume that files are an array.

After trying to find a more concise way of generating the list of files, I just realized that the check is not working. The check still passes if I add a tab to a shell script on the main branch. I'm working on rewriting the implementation.

Reply:
> After trying to find a more concise way of generating the list of files, I just realized that the check is not working.

Do you mean the existing script doesn't work either?

Reply:
This check is not working. Refer to pull request: https://github.com/etcd-io/etcd/pull/19049

And the false positive result from prow: https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/etcd-io_etcd/19049/pull-etcd-verify/1867302228360957952#1:build-log.txt%3A504-511

However, in the end, the `pull-etcd-verify` job fails due to the `fix` step, which replaces tabs with whitespace, and it leaves modified files in the git repository (https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/etcd-io_etcd/19049/pull-etcd-verify/1867302228360957952#1:build-log.txt%3A919-925)

@ahrtr, WDYT, should we fix the check or delete it and keep it covered by `fix`? I don't have a strong opinion, but I lean toward deleting the check to avoid duplication.

Reply:
Thanks for the confirmation. I think that we still need to fix it. `verify` (including `verify-shellws`) is used to verification, and `fix` is used to fix the issues detected by `verify`. Both of them should work.

Reply:
Done. Expected failure with the revised code: https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/etcd-io_etcd/19049/pull-etcd-verify/1867363984722628608